src/components/NotificationFeed/NotificationFeed.tsx

```diff
@@ -50,19 +56,16 @@ export const NotificationFeed: React.FC<NotificationFeedProps> = ({
   onNotificationClick,
   onMarkAllAsReadClick,
 }) => {
-  const {
-    status,
-    setStatus,
-    feedClient,
-    useFeedStore,
-    colorMode,
-  } = useKnockFeed();
-
-  const pageInfo = useFeedStore((state) => state.pageInfo);
-  const items = useFeedStore((state) => state.items);
-  const networkStatus = useFeedStore((state) => state.networkStatus);
+  const [status, setStatus] = useState(FilterStatus.All);
+  const { feedClient, useFeedStore, colorMode } = useKnockFeed();
 
+  const { pageInfo, items, networkStatus } = useFeedStore();
   const containerRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    feedClient.fetch({ status });
```

Comment:
When the status changes, refetch the feed (used to be in the `KnockFeedProvider`)